packages/opentelemetry-web/src/WebTracer.ts

```diff
@@ -14,25 +14,51 @@
  * limitations under the License.
  */
 
-import { BasicTracer, BasicTracerConfig } from '@opentelemetry/tracing';
+import { BasePlugin } from '@opentelemetry/core';
+import {
+  BasicTracer,
+  BasicTracerConfig,
+  SpanProcessor,
+} from '@opentelemetry/tracing';
 import { StackScopeManager } from './StackScopeManager';
 
+/**
+ * WebTracerConfig provides an interface for configuring a Web Tracer.
+ */
+export interface WebTracerConfig extends BasicTracerConfig {
+  /**
+   * span processor
+   */
+  spanProcessor?: SpanProcessor;
+  /**
+   * plugins to be used with tracer, they will be enabled automatically
+   */
+  plugins?: BasePlugin<unknown>[];
+}
+
 /**
  * This class represents a web tracer with {@link StackScopeManager}
  */
 export class WebTracer extends BasicTracer {
   /**
    * Constructs a new Tracer instance.
+   * @param {WebTracerConfig} config Web Tracer config
    */
-  /**
-   *
-   * @param {BasicTracerConfig} config Web Tracer config
-   */
-  constructor(config: BasicTracerConfig = {}) {
+  constructor(config: WebTracerConfig = {}) {
     if (typeof config.scopeManager === 'undefined') {
       config.scopeManager = new StackScopeManager();
     }
-    config.scopeManager.enable();
+    if (typeof config.plugins === 'undefined') {
+      config.plugins = [];
```

Comment:
Instead of empty array (`[]`), do you think we should load all avaiable plugin by default. Similar to NodeTracer -> https://github.com/open-telemetry/opentelemetry-js/blob/master/packages/opentelemetry-node/src/NodeTracer.ts#L39?

Reply:
I think @draffensperger was against that already. In node it doesn't matter if you keep plugins as part of node, but in web you don't want to bundle them if someone doesn't want to use them explicitly.

Reply:
Yes, so with web clients, a big concern is the size of the JS bundle that gets sent to the client. The bigger that bundle, the slower the page will load, especially on mobile phones with slow bandwidth and lower CPU to parse the JavaScript.

So I think our web tracer should have no plugins by default. Unlike Node, where you intercept the plugins using require-in-the-middle, I think it's better with the web to explicitly import them and pass them to the tracer. That works better for bundlers, compilers like webpack, rollup, Closure Compiler, etc. that may inline things and do a bunch of changes to the code before spitting out JS for the browser.

Does that sound OK?

We could potentially have something like a set of recommended plugins and even fully compiled JS files for easy use in `<script>` tags, but I would think those should live in a separate package(s).

Reply:
@mayurkale22 and @draffensperger so whats the conclusion, personally I think the same as for web plugins should be loaded only those that were specified directly not like defaults in node.

Reply:
@draffensperger thanks for the explanation, I agree with your point.